FTBFS on 24.10: `‘MEM[(struct function &)&D.24324 + 112]._M_invoker’ may be used uninitialized`

```
In member function ‘__ct ’,
    inlined from ‘_Construct’ at /usr/include/c++/14/bits/stl_construct.h:119:7,
    inlined from ‘_M_construct’ at /usr/include/c++/14/optional:274:19,
    inlined from ‘__ct ’ at /usr/include/c++/14/optional:151:22,
    inlined from ‘__ct ’ at /usr/include/c++/14/optional:147:7,
    inlined from ‘__ct_base ’ at /usr/include/c++/14/optional:395:42,
    inlined from ‘__ct ’ at /usr/include/c++/14/optional:429:57,
    inlined from ‘__ct ’ at /usr/include/c++/14/optional:542:9,
    inlined from ‘__ct ’ at /usr/include/c++/14/optional:703:11,
    inlined from ‘__ct ’ at /usr/src/wlcs-1.7.0+dev91-g7cb0947-0ubuntu24.10/tests/xdg_popup.cpp:132:8,
    inlined from ‘Values’ at /usr/include/gtest/gtest-param-test.h:336:20,
    inlined from ‘gtest_DefaultXdgPopupPositionerTest_EvalGenerator_’ at /usr/src/wlcs-1.7.0+dev91-g7cb0947-0ubuntu24.10/tests/xdg_popup.cpp:733:1:
/usr/include/c++/14/bits/std_function.h:405:42: error: ‘MEM[(struct function &)&D.24324 + 112]._M_invoker’ may be used uninitialized [-Werror=maybe-uninitialized]
  405 |       : _Function_base(), _M_invoker(__x._M_invoker)
      |                                          ^
/usr/src/wlcs-1.7.0+dev91-g7cb0947-0ubuntu24.10/tests/xdg_popup.cpp: In function ‘gtest_DefaultXdgPopupPositionerTest_EvalGenerator_’:
/usr/src/wlcs-1.7.0+dev91-g7cb0947-0ubuntu24.10/tests/xdg_popup.cpp:733:1: note: ‘<anonymous>’ declared here
```

--- a/debian/rules
+++ b/debian/rules
@@ -26,12 +26,18 @@ endif
 
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105329
 ifeq ($(DEB_HOST_ARCH), ppc64el)
-  ifneq ($(shell gcc --version | grep '1[23].[[:digit:]]\+.[[:digit:]]\+$$'),)
+  ifneq ($(shell gcc --version | grep '1[34].[[:digit:]]\+.[[:digit:]]\+$$'),)
     export DEB_CFLAGS_MAINT_APPEND = -O2
     export DEB_CXXFLAGS_MAINT_APPEND = -O2
     export DEB_FCFLAGS_MAINT_APPEND = -O2
     export DEB_FFLAGS_MAINT_APPEND = -O2
   endif
+endif
+
+# Disable LTO on broken binutils
+# https://bugs.launchpad.net/ubuntu/+source/binutils/+bug/2070302
+ifneq ($(shell ld --version | grep '2.43.1$$'),)
+	DEB_BUILD_MAINT_OPTIONS += optimize=-lto
 endif
 
 override_dh_auto_configure:

